### PR TITLE
Prevent hovering / interacting with hidden stuff

### LINKED ISSUE #1063: Hovering works for things hidden behind FoV
# Preliminary information

## Category
BUG

## Bug: Release or Development?
develop

# Report
## Current Behavior
Hovering mouse over the black (FoV-covered) zone shows tooltips for these objects

## Expected/Wanted/Requested Behavior
Hovering shouldn't work for things you can't see

## Steps to Reproduce
Please enter the steps to reproduce the bug or behaviour:
fovtooltips

## OS used:
Windows

## Network
Self-hosted/steam server

### LINKED ISSUE #1426: Can interact with hidden wallmounts
# Description

## Category
TODO

## Steps to Reproduce
Please enter the steps to reproduce the bug or behaviour:
1. Find a lightswitch in the room, remember its position
2. Get outside the room, stand next to the tile that has the lightswitch on the other side
3. Click the spot where the lightswitch is located

scr 2019-01-08 14 04 14 2019-01-08 14_21_12

A serverside check should be in place to refuse object interaction if player is in the range but is not facing(seeing?) wallmount

### PR DESCRIPTION
**Pull request by chairbender**
### Purpose
Fixes #1063 
Fixes #1426 

Prevents interaction with FOV-occluded stuff as well as wallmounts on the wrong side of the wall. Also adds server-side checks for interacting with wrong side of wallmount.

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  This PR has been tested in multiplayer

### CODE REVIEW

File: UnityProject/Assets/Scripts/UI/Right click/Rightclick.cs [hunk 1]
```diff
@@ -54,82 +61,142 @@ public class Menu {
...
-	void Update () {
-		if (Input.GetMouseButtonDown (1)) {
-			Vector3 position = Camera.main.ScreenToWorldPoint(UnityEngine.Input.mousePosition);
-			position.z = 0f;
-			//gets Items on the position of the mouse
-			List<GameObject> objects = UITileList.GetItemsAtPosition(position);
+	void Update()
+	{
+		//do nothing if the mouse is over something occluded
+		if (lightingSystem.IsFovOccluded(Input.mousePosition))
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/UI/Right click/Rightclick.cs:102]: this is one of the actual changes in this file

File: UnityProject/Assets/Scripts/UI/Right click/Rightclick.cs [hunk 2]
```diff
@@ -54,82 +61,142 @@ public class Menu {
...
 			//Logger.Log ("yo", Category.UI);
-			if (options.Count > 0){
+			if (options.Count > 0)
+			{
 				RadialMenuSpawner.ins.SpawnRadialMenu(options);
 			}
+		}
+	}
+
+	private List<GameObject> GetRightClickableObjects()
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/UI/Right click/Rightclick.cs:120]: this is one of the actual changes in this file

File: UnityProject/Assets/Scripts/UI/Right click/Rightclick.cs [hunk 3]
```diff
@@ -54,82 +61,142 @@ public class Menu {
...
+		position.z = 0f;
+		List<GameObject> objects = UITileList.GetItemsAtPosition(position);
+
+		//special case, remove wallmounts that are transparent
+		objects.RemoveAll(IsHiddenWallmount);
+
+		return objects;
+	}
 
+	private bool IsHiddenWallmount(GameObject obj)
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/UI/Right click/Rightclick.cs:132]: this is one of the actual changes in this file

**Comment by DooblyNoobly** [comment]
nice! works well. Thanks

**Pull request merged**